src/Build/BackEnd/Components/RequestBuilder/TaskBuilder.cs

```diff
@@ -155,6 +155,9 @@ public async Task<WorkUnitResult> ExecuteTask(TargetLoggingContext loggingContex
             ErrorUtilities.VerifyThrow(taskInstance != null, "Need to specify the task instance.");
 
             _buildRequestEntry = requestEntry;
+
+            _taskExecutionHost.TaskEnvironment = requestEntry.TaskEnvironment;
```

Comment:
The fields in taskExecutionHost are populated through InitializeForTask
please use it instead to init TaskEnvironment